src/i18n.ts

```diff
@@ -0,0 +1,248 @@
+import { Context, Fluent, FluentType, resolve } from "./deps.ts";
+
+import type {
+  FluentBundleOptions,
+  FluentValue,
+  LocaleId,
+  Middleware,
+  NextFunction,
+  Scope,
+  TranslationContext,
+} from "./deps.ts";
+
+import { exists, readLocalesDir } from "./utils.ts";
+
+import type {
+  I18nConfig,
+  I18nContextFlavor,
+  TranslateFunction,
+} from "./types.ts";
+
+class FluentContext extends FluentType<string> {
+  constructor(value: string, private defaultValue = "") {
+    super(value);
+  }
+
+  toString(scope: Scope): string {
+    let ctx = JSON.parse(scope.args?.ctx as string);
+    const keys = this.value.split(".");
+    for (const key of keys) {
+      ctx = ctx[key];
+      if (ctx === undefined) break;
+    }
+    return ctx ?? this.defaultValue;
+  }
+}
+
+/**
+ * A custom function for accessing `Context` properties directly from
+ * Fluent translation files.
+ */
+export function CTX(args: FluentValue[]) {
+  if (args[1] !== undefined && typeof args[1] !== "string") {
```

Comment:
Avoid redundant check.
```suggestion
  if (typeof args[1] !== "string") {
```

Reply:
Fixed